From cdbf61a3c4e647199e6f575a3570885f4e54c41b Mon Sep 17 00:00:00 2001
From: Yu Liao <liaoyu15@huawei.com>
Date: Fri, 31 Dec 2021 15:59:18 +0800
Subject: [PATCH] kabi: reserve space for struct worker
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: cdbf61a3c4e647199e6f575a3570885f4e54c41b


hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I4JBL0
CVE: NA

-------------------------------

Reserve space for struct worker.

Signed-off-by: Yu Liao <liaoyu15@huawei.com>
Reviewed-by: Cheng Jian <cj.chengjian@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 kernel/workqueue_internal.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/kernel/workqueue_internal.h b/kernel/workqueue_internal.h
index 498de0e909a4..9f97a95190a1 100644
--- a/kernel/workqueue_internal.h
+++ b/kernel/workqueue_internal.h
@@ -11,6 +11,7 @@
 #include <linux/workqueue.h>
 #include <linux/kthread.h>
 #include <linux/preempt.h>
+#include <linux/kabi.h>
 
 struct worker_pool;
 
@@ -57,6 +58,11 @@ struct worker {
 
 	/* used by the scheduler to determine a worker's last known identity */
 	work_func_t		last_func;
+
+	KABI_RESERVE(1)
+	KABI_RESERVE(2)
+	KABI_RESERVE(3)
+	KABI_RESERVE(4)
 };
 
 /**
-- 
2.26.2

